src/tests/ftest/util/daos_utils.py

```diff
@@ -377,39 +375,83 @@ def container_create_snap(self, pool, cont, snap_name=None, epoch=None,
+        # Sample create-snap output.
+        # snapshot/epoch 1582610056530034697 has been created
+        data = {}
+        match = re.findall(
+            r"[A-Za-z\/]+\s([0-9]+)\s[a-z\s]+", self.result.stdout)
+        if match:
+            data["epoch"] = match[0]
+        return data
+
+    def container_destroy_snap(self, pool, cont, snap_name=None, epc=None,
+                               svc=None, sys_name=None, epcrange=None):
+        """Call daos container destroy-snap.
 
         Args:
             pool (str): Pool UUID.
             cont (str): Container UUID.
             snap_name (str, optional): Snapshot name.
-            epoch (str, optional): Epoch number.
+            epc (str): Epoch value of the snapshot to be destroyed.
             svc (str, optional): Pool service replicas, e.g., '1,2,3'. Defaults
                 to None.
             sys_name (str, optional): DAOS system name context for servers.
                 Defaults to None.
+            epcrange (list): If you want to use --epc, use a single-element
+                list. If you want to use --epcrange, use a list that contains
+                epoch for B in the first element and epoch for E in the second
+                element.
 
         Returns:
             CmdResult: Object that contains exit status, stdout, and other
                 information.
 
         Raises:
-            CommandFailure: if the daos pool query command fails.
+            CommandFailure: if the daos container destroy-snap command fails.
 
         """
-        return self._get_result(
-            ("container", "destroy-snap"), pool=pool, svc=svc, cont=cont,
-            sys_name=sys_name, snap=snap_name, epc=epoch, )
+        kwargs = {
+            "pool": pool,
+            "svc": svc,
+            "cont": cont,
+            "sys_name": sys_name,
+            "snap": snap_name,
+            "epc": epc,
+            "epcrange": epcrange
+        }
+        return self._get_result(("container", "destroy-snap"), **kwargs)
+
+    def container_list_snaps(self, pool, cont, svc=None):
+        """List snapshot in a container.
+
+        Args:
+            pool (str): Pool UUID.
+            cont (str): Container UUID.
+            svc (str): Service replicas. Defaults to None.
+
+        Returns:
+            dict: Dictionary that contains epoch values in key "epochs". Value
+                is a list of string.
+        """
+        self.result = self._get_result(
+            ("container", "list-snaps"), pool=pool, cont=cont, svc=svc)
+
+        # Sample container list-snaps output - no line break on second line.
+        # 05/15-20:40:40.46 wolf-3 Container's snapshots :
+        # 05/15-20:40:40.46 wolf-3 1589575199653691408  05/15-20:40:40.46
+        # wolf-3 1589575236150689813  05/15-20:40:40.46 wolf-3
+        data = {}
+        match = re.findall(r"(\d{19})", self.result.stdout)
+        if match:
+            data["epochs"] = match
```

Comment:
The sample output would seem to indicate that it could list either one snapshot across multiple servers or multiple snapshots across one or more servers.  If that is possible, shouldn't this be taken into account in parsing the output?  Should we also capture the time stamps?  Do we need to verify the time stamp information in the test?

Reply:
I checked the list-snaps output on the latest master and it prints only the epochs, so I think we can keep the code as it is.

Reply:
Thanks for checking.